UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -370,33 +444,35 @@ public async Task<BatchStatus> CommitAndGetBatchStatus(string batchId, string co
                 var batchStatusMetaData = new BatchStatusMetaData()
                 {
                     AccessToken = accessToken,
-                    BatchId = batchId
+                    BatchId = batchId,
+                    FileName = customFileInfo.Name
                 };
                 Stopwatch watch = new Stopwatch();
                 watch.Start();
                 while (batchStatus != BatchStatus.Committed && watch.Elapsed.TotalMinutes <= fileShareServiceConfig.Value.BatchCommitCutOffTimeInMinutes)
-                {                  
+                {
                     batchStatus = await GetBatchStatus(batchStatusMetaData, correlationId);
                     if (batchStatus == BatchStatus.Failed)
                     {
                         watch.Stop();
                         logger.LogError(EventIds.BatchFailedStatus.ToEventId(), "Batch status failed for BatchId {batchId} and _X-Correlation-ID:{CorrelationId}", batchStatusMetaData.BatchId, correlationId);
-                        return batchStatus;
+                        throw new FulfilmentException(EventIds.BatchFailedStatus.ToEventId());
+
                     }
-                 await Task.Delay(fileShareServiceConfig.Value.BatchCommitDelayTimeInMilliseconds);
+                    await Task.Delay(fileShareServiceConfig.Value.BatchCommitDelayTimeInMilliseconds);
                 }
                 if (batchStatus != BatchStatus.Committed)
                 {
                     watch.Stop();
                     logger.LogError(EventIds.BatchCommitTimeout.ToEventId(), "Batch Commit Status timedout with BatchStatus:{batchStatus} for BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", batchStatus, batchStatusMetaData.BatchId, correlationId);
-                    return batchStatus;
+                    throw new FulfilmentException(EventIds.BatchCommitTimeout.ToEventId());
                 }
                 watch.Stop();
             }
             else
-            {
-                batchStatus = BatchStatus.Failed;
+            {               
                 logger.LogError(EventIds.BatchFailedStatus.ToEventId(), "Batch status failed for BatchId {batchId} and _X-Correlation-ID:{CorrelationId}", batchId, correlationId);
+                throw new FulfilmentException(EventIds.BatchFailedStatus.ToEventId());
```

Comment:
Can we log filename as well in all logs of CommitAndGetBatchStatus and UploadFileToFileShareService methods as it is commonly used for both V01X01.zip and error.txt files

Reply:
Added details. Resolved